waltz-client/src/main/java/com/wepay/waltz/client/internal/Partition.java

```diff
@@ -269,17 +271,22 @@ public void partitionAhead() {
     public void ensureMounted() {
         if (!mounted) {
             synchronized (lock) {
+                final long due = System.currentTimeMillis() + ENSURE_MOUNTED_TIMEOUT;
                 while (state != PartitionState.CLOSED && !mounted) {
                     if (transactionMonitor.isStopped()) {
                         throw new PartitionInactiveException(partitionId);
                     } else if (clientHighWaterMarkAhead) {
                         throw new IllegalStateException(String.format("client is ahead of store for partition: %d", partitionId));
                     }
-
-                    try {
-                        lock.wait();
-                    } catch (InterruptedException ex) {
-                        Thread.interrupted();
+                    long remaining = due - System.currentTimeMillis();
+                    if (remaining > 0) {
+                        try {
+                            lock.wait(ENSURE_MOUNTED_TIMEOUT);
+                        } catch (InterruptedException ex) {
+                            Thread.interrupted();
+                        }
+                    } else {
+                        throw new WaitForMountedTimeoutException(partitionId, ENSURE_MOUNTED_TIMEOUT);
```

Comment:
Optional suggestion: passing a variable for logging only may not be best practice (even if I see it's been done in Waltz). The reason is it may implicitly suggests that the variable is needed by the class itself for some additional processing. Passing a log message to the Exception class may look cleaner